packages/xod-func-tools/src/monads.js

```diff
@@ -116,6 +116,17 @@ export const eitherToPromise = foldEither(
   Promise.resolve.bind(Promise)
 );
 
+// :: (() -> b) -> (a -> a) -> Maybe a -> Promise a b
+export const maybeToPromise = R.curry(
+  (nothingFn, justFn, maybe) => new Promise(
+    (resolve, reject) => (
+      maybe.isJust ? resolve(maybe.value) : reject()
+    )
+  )
+    .catch(nothingFn)
+    .then(justFn)
```

Comment:
`justFn` will always run, even if `maybe` is a `Just`

Reply:
You mean is a `Nothing`? :)
Yep, if in `nothingFn` we return resolved promise...
I don't know how to handle it in another way (calling them inside `resolve()` or `reject()` functions makes nested promises 🤢 ).

Reply:
> You mean is a Nothing? :)
Yes, sorry :)

Reply:
> Yep, if in nothingFn we return resolved promise...

![screen shot 2017-11-23 at 19 53 17](https://user-images.githubusercontent.com/2527093/33183250-2404db82-d088-11e7-9342-865fcadf36d1.png)